internal/testutil/mig/test_case.go

```diff
@@ -0,0 +1,60 @@
+package mig
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/stretchr/testify/require"
+)
+
+// ConvertToMigrationTest returns an updated TestCase that reuses step 1 and adds a TestStepCheckEmptyPlan
+// Requires: `MONGODB_ATLAS_LAST_VERSION` to be present
+func ConvertToMigrationTest(t *testing.T, test *resource.TestCase) resource.TestCase {
+	t.Helper()
+	validateReusableCase(t, test)
+	firstStep := test.Steps[0]
+	steps := []resource.TestStep{
+		useExternalProvider(&firstStep, ExternalProviders()),
+		TestStepCheckEmptyPlan(firstStep.Config),
+	}
+	return reuseCase(test, steps)
+}
+
+// ConvertToMigrationTestUseExternalProvider returns an updated TestCase that reuses step 1 and adds a TestStepCheckEmptyPlan with the additionalProviders
+// Requires: `MONGODB_ATLAS_LAST_VERSION` to be present
+// externalProviders: e.g., ExternalProvidersWithAWS() or ExternalProviders("specific_sem_ver")
+// additionalProviders: e.g., acc.ExternalProvidersOnlyAWS(), can also be nil
+func ConvertToMigrationTestUseExternalProvider(t *testing.T, test *resource.TestCase, externalProviders, additionalProviders map[string]resource.ExternalProvider) resource.TestCase {
```

Comment:
Example of users will be: `resource_privatelink_endpoint_service_migration_test.go` and `resource_encryption_at_rest_migration_test.go`

Reply:
moved this to the PR description